lib/src/results.dart

```diff
@@ -207,3 +210,30 @@ extension RealmResultsInternal on RealmResults {
     return RealmResults<T>._(handle, realm);
   }
 }
+
+class RealmResultsIterator<T extends RealmObject> implements Iterator<T> {
+  final Iterable<T> _iterable;
+  final int _length;
+  int _index;
+
+  RealmResultsIterator(Iterable<T> iterable)
```

Comment:
This ctor should take a `RealmResults<T>` - it will not work for an arbitrary `Iterable<T>`. Also, consider making it a private class - users should not be instantiating these by them selves.

Reply:
Fixed in https://github.com/realm/realm-dart/pull/153/commits/86e68144180eff8d4aa10ca61940ee4dc50c4b8b
and https://github.com/realm/realm-dart/pull/153/commits/9c67ce3f8dd5edd209001a2f6f07e784962bc593